packages/itmat-interface/src/lxd/lxdManager.ts

```diff
@@ -0,0 +1,401 @@
+                return {
+                    error: true,
+                    data: `Failed to fetch profile data. ${response.data}`
+                };
+            }
+        } catch (error: any) {
+            if (error.response) {
+                console.error(`Failed to fetch profile data. ${error.response}`);
+                return {
+                    error: true,
+                    data: `Failed to fetch profile data. ${error.response}`
+                };
+            } else {
+                console.error('Error fetching profile data from LXD:', error);
+                return {
+                    error: true,
+                    data: error
+                };
+            }
+        }
+    },
+
+    // This should almost never be run
+    getInstances: async () => {
+        try {
+            const instanceUrls = await lxdInstance.get('/1.0/instances');
+            const instances = await Promise.allSettled(instanceUrls.data.metadata.map(async (instanceUrl: string) => await lxdInstance.get(instanceUrl)));
+
+            const sanitizedIntances = instances.map((instance: any) => {
+
+                const { metadata } = instance.value.data;
+                return {
+                    name: metadata.name,
+                    description: metadata.description,
+                    status: metadata.status,
+                    statusCode: metadata.status_code,
+                    profiles: metadata.profiles,
+                    type: metadata.type,
+                    architecture: metadata.architecture,
+                    creationDate: metadata.created_at,
+                    lastUsedDate: metadata.last_used_at,
+                    // config: metadata.config
+                    username: metadata.config['user.username'] || 'N/A',
+                    cpuLimit: metadata.config['limits.cpu'] || 'N/A',
+                    memoryLimit: metadata.config['limits.memory'] || 'N/A'
+                };
+            });
+            return {
+                data: sanitizedIntances
+            };
+        } catch (e) {
+            if (isAxiosError(e)) {
+                console.error('getInstances axios error', e.message);
+                return {
+                    error: true,
+                    data: e.message
+                };
+            }
+            console.error('getInstances unknown error 1', e);
+            return {
+                error: true,
+                data: e
+            };
+        }
+    },
+    getInstanceInfo: async (instanceName: string) => {
+        instanceName = encodeURIComponent(instanceName);
+        return await lxdInstance.get(`/1.0/instances/${instanceName}`);
+    },
+    getInstanceState: async (instanceName: string) => {
+        instanceName = encodeURIComponent(instanceName);
+        return await lxdInstance.get(`/1.0/instances/${instanceName}/state`);
+
+    },
+    // getJupyterServiceUrl
+    getJupyterServiceUrl: async (instanceName: string) => {

```

Comment:
This probably shouldn't be here, this has nothing to do with this class `LXDManager`. Also there are two call to `/state` API, that's not the better approach.